pvnet/models/multimodal/config_validation.py

```diff
@@ -0,0 +1,1007 @@
+        expected_type=int,
+        warn_on_type_mismatch=True,
+    )
+    _check_key(cfg, "nwp_interval_minutes", required=False, expected_type=dict)
+    _check_key(cfg, "embedding_dim", required=False, expected_type=int)
+    _check_key(cfg, "include_sun", required=False, expected_type=bool)
+    _check_key(cfg, "include_gsp_yield_history", required=False, expected_type=bool)
+    _check_key(cfg, "add_image_embedding_channel", required=False, expected_type=bool)
+
+    # Check required sections are dicts with targets
+    _check_dict_section(cfg, "output_network", required=True, check_target=True)
+    _check_dict_section(cfg, "optimizer", required=True, check_target=True)
+
+    # Satellite Encoder
+    sat_section = _check_dict_section(
+        cfg, "sat_encoder", required=False, check_target=True
+    )
+    _check_time_parameter(
+        cfg,
+        "sat_history_minutes",
+        owner_key="sat_encoder",
+        required_if_owner_present=True,
+    )
+    if sat_section:
+        _check_convnet_encoder_params(
+            cfg, section_key="sat_encoder", context="sat_encoder"
+        )
+
+    # PV Encoder
+    pv_section = _check_dict_section(
+        cfg, "pv_encoder", required=False, check_target=True
+    )
+    _check_time_parameter(
+        cfg,
+        "pv_history_minutes",
+        owner_key="pv_encoder",
+        required_if_owner_present=True,
+    )
+    if pv_section:
+        _check_attention_encoder_params(
+            cfg, section_key="pv_encoder", context="pv_encoder"
+        )
+
+    # NWP Encoders
+    nwp_section = _check_dict_section(
+        cfg, "nwp_encoders_dict", required=False, check_sub_items_target=True
+    )
+    if nwp_section is not None:
+        _validate_nwp_specifics(cfg, nwp_section)
+        for source in nwp_section.keys():
+            _check_convnet_encoder_params(
+                cfg=cfg,
+                section_key="nwp_encoders_dict",
+                context=f"nwp_encoders_dict[{source}]",
+                source_key=source,
+            )
+
+
+def validate(
+    numpy_batch: NumpyBatch,
+    multimodal_config: dict,
+    input_data_config: dict,
+    expected_batch_size: Optional[int] = None,
+) -> None:
+    """Validate a batch of numpy data against the multimodal configuration.
+
+    Performs static config validation first, then checks shapes of data arrays
+    in the batch against expectations derived from the config and input_data_config.
+
+    Args:
+        numpy_batch: Dictionary containing modality data as NumPy arrays.
+        multimodal_config: The multimodal configuration dictionary (for model settings).
+        input_data_config: The input data configuration dictionary (for data source settings).
+        expected_batch_size: Optional expected batch size to validate against.
+
+    Raises:
+        KeyError, TypeError, ValueError: If config is invalid or batch shapes mismatch config.
+    """
+
+    try:
+        _validate_static_config(multimodal_config)
```

Comment:
could intitla the model here. As long as it doesnt take to long / use too much memory. There might be less code to maintain then